test/functional/apps/discover/_discover.ts

```diff
@@ -19,328 +19,369 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
-      log.debug('load kibana index with default index pattern');
-      await kibanaServer.importExport.load('test/functional/fixtures/kbn_archiver/discover');
-      // and load a set of makelogs data
-      await esArchiver.loadIfNeeded('test/functional/fixtures/es_archiver/logstash_functional');
-      await kibanaServer.uiSettings.replace(defaultSettings);
-      await PageObjects.common.navigateToApp('discover');
-      await PageObjects.timePicker.setDefaultAbsoluteRange();
-    });
-    after(async () => {
-      await kibanaServer.savedObjects.clean({ types: ['search', 'index-pattern'] });
-    });
-    describe('query', function () {
-      const queryName1 = 'Query # 1';
-
-      it('should show correct time range string by timepicker', async function () {
-        const time = await PageObjects.timePicker.getTimeConfig();
-        expect(time.start).to.be(PageObjects.timePicker.defaultStartTime);
-        expect(time.end).to.be(PageObjects.timePicker.defaultEndTime);
-        const rowData = await PageObjects.discover.getDocTableIndex(1);
-        log.debug('check the newest doc timestamp in UTC (check diff timezone in last test)');
-        expect(rowData).to.contain('Sep 22, 2015 @ 23:50:13.253');
+    describe('discover accessibility tests', () => {
+      before(setUpDiscoverPage);
+      after(tearDownDiscoverPage);
+
+      it('should give focus to the saved search title h1 on navigate', async () => {
+        const skipButton = await testSubjects.find('discoverSavedSearchTitle');
+        const activeElement = await find.activeElement();
+        expect(await skipButton.getAttribute('data-test-subj')).to.eql(
+          await activeElement.getAttribute('data-test-subj')
+        );
       });
 
-      it('save query should show toast message and display query name', async function () {
-        await PageObjects.discover.saveSearch(queryName1);
-        const actualQueryNameString = await PageObjects.discover.getCurrentQueryName();
-        expect(actualQueryNameString).to.be(queryName1);
+      it('should give focus to the data view switch link when Tab is pressed', async () => {
+        await browser.pressKeys(browser.keys.TAB);
+        const dataViewSwitchLink = await testSubjects.find('discover-dataView-switch-link');
+        const activeElement = await find.activeElement();
+        expect(await dataViewSwitchLink.getAttribute('data-test-subj')).to.eql(
+          await activeElement.getAttribute('data-test-subj')
+        );
       });
+    });
 
-      it('load query should show query name', async function () {
-        await PageObjects.discover.loadSavedSearch(queryName1);
-
-        await retry.try(async function () {
-          expect(await PageObjects.discover.getCurrentQueryName()).to.be(queryName1);
-        });
+    describe('discover functionality tests', () => {
+      before(async function () {
+        await setUpDiscoverPage();
+        await PageObjects.timePicker.setDefaultAbsoluteRange();
```

Comment:
This call before each test caused my accessibility tests to fail since it changes the initial focus in the browser. To fix this I broke the tests up into 'accessibility tests' and 'functionality tests' so I could skip running this function in the before hook. I'm open to any suggestions about a better way to handle this though.

Reply:
We could also move "discover accessibility tests" out to a new separate file and don't make this file even more complex.

Reply:
Good idea, I'll pull these into a separate `_discover_accessibility.ts` file and update the PR.